ogzaf/demographics.py

```diff
@@ -1,96 +1,352 @@
+            "SexId": "sex_num",
+            "Sex": "sex_str",
+            "AgeStart": "age",
+            "Value": "mort_rate",
+        },
+        inplace=True,
+    )
+
+    # Clean the data
+    infmort_rate_df["infmort_rate"] = (
+        infmort_rate_df["inf_deaths_p_1000"] / 1000
+    )
+
+    infmort_rate_df = infmort_rate_df[
+        (
+            (infmort_rate_df["year"] >= start_year)
+            & (infmort_rate_df["year"] <= end_year)
+        )
+    ]
+    mort_rates_df = mort_rates_df[
+        (
+            (mort_rates_df["year"] >= start_year)
+            & (mort_rates_df["year"] <= end_year)
+        )
+    ]
+
+    return infmort_rate_df, mort_rates_df
+
+
+def get_un_pop_data(
+    country_id: str = "710",
+    start_year: int = 2021,
+    end_year: int = None,
+    download: bool = True,
+) -> pd.DataFrame:
+    """
+    Get UN population data for a country for some range of years (at least
+    one year) and by age. The country_id=710 is for South Africa. These data
+    come from the United Nations Data Portal API for UN population data (see
+    https://population.un.org/dataportal/about/dataapi)
+
+    Args:
+        country_id (str): 3-digit country id (numerical)
+        start_year (int): beginning year of the data
+        end_year (int): end year of the data
+        download (bool): whether to download the data from the UN Data Portal.
+            If False, a path must be specified in the path_folder argument.
+        path_folder (None or str): string path to folder where data are stored
+
+    Returns:
+        fert_rates_df (DataFrame): dataset with fertility rates by age
+    """
+    if end_year is None:
+        end_year = start_year
+    # UN variable code for Population by 1-year age groups and sex
+    pop_code = "47"
+
+    if download:
+        pop_target = (
+            "https://population.un.org/dataportalapi/api/v1/data/indicators/"
+            + pop_code
+            + "/locations/"
+            + country_id
+            + "/start/"
+            + str(start_year)
+            + "/end/"
+            + str(end_year)
+            + "?format=csv"
+        )
+    else:
+        pop_target = os.path.join(DATA_DIR, "un_zaf_pop.csv")
+
+    # Convert .csv file to Pandas DataFrame
+    pop_df = pd.read_csv(
+        pop_target,
+        sep="|",
+        header=1,
+        usecols=["TimeLabel", "SexId", "Sex", "AgeStart", "Value"],
+        float_precision="round_trip",
+    )
+
+    # Rename variables in the population and fertility rates data
+    pop_df.rename(
+        columns={
+            "TimeLabel": "year",
+            "SexId": "sex_num",
+            "Sex": "sex_str",
+            "AgeStart": "age",
+            "Value": "pop",
+        },
+        inplace=True,
+    )
+
+    # Clean the data
+    pop_df = pop_df[
+        ((pop_df["year"] >= start_year) & (pop_df["year"] <= end_year))
+    ]
+
+    return pop_df
 
 
 def get_fert(totpers, min_yr, max_yr, graph=False):
```

Comment:
```suggestion
def get_fert(min_yr, max_yr, graph=False):
```
`totpers` is no longer used in the revised function